plugins/outputs/postgresql/postgresql_bench_test.go

```diff
@@ -0,0 +1,90 @@
+package postgresql
+
+import (
+	"context"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+)
+
+func BenchmarkPostgresql_sequential(b *testing.B) {
+	gen := batchGenerator(ctx, b, 1000, 3, 8, 12, 100, 2)
+	benchmarkPostgresql(b, gen, 1, true)
+}
+func BenchmarkPostgresql_concurrent(b *testing.B) {
+	gen := batchGenerator(ctx, b, 1000, 3, 8, 12, 100, 2)
+	benchmarkPostgresql(b, gen, 10, true)
+}
+
+func benchmarkPostgresql(b *testing.B, gen <-chan []telegraf.Metric, concurrency int, foreignTags bool) {
+	p := newPostgresqlTest(b)
+	p.Connection += fmt.Sprintf(" pool_max_conns=%d", concurrency)
+	p.TagsAsForeignKeys = foreignTags
+	p.LogLevel = ""
+	if err := p.Connect(); err != nil {
+		b.Fatalf("Error: %s", err)
+	}
+
+	metricCount := 0
+
+	b.ResetTimer()
+	tStart := time.Now()
+	for i := 0; i < b.N; i++ {
+		batch := <-gen
+		if err := p.Write(batch); err != nil {
+			b.Fatalf("Error: %s", err)
+		}
+		metricCount += len(batch)
+	}
+	p.Close()
+	b.StopTimer()
+	tStop := time.Now()
+	b.ReportMetric(float64(metricCount) / tStop.Sub(tStart).Seconds(), "metrics/s")
+}
+
+// tagCardinality counts all the tag keys & values as one element. fieldCardinality counts all the field keys (not values) as one element.
+func batchGenerator(ctx context.Context, b *testing.B, batchSize int, numTables int, numTags int, numFields int, tagCardinality int, fieldCardinality int) <-chan []telegraf.Metric {
```

Comment:
Please define a struct and pass the values through this struct to avoid the linter error.